crates/sui-rpc-api/proto/sui/rpc/v2alpha/effects.proto

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package sui.rpc.v2alpha;
+
+import "sui/rpc/v2alpha/bcs.proto";
+import "sui/rpc/v2alpha/execution_status.proto";
+import "sui/rpc/v2alpha/gas_cost_summary.proto";
+import "sui/rpc/v2alpha/owner.proto";
+import "sui/rpc/v2alpha/version.proto";
+
+// The effects of executing a transaction.
+message TransactionEffects {
+  // This TransactionEffects serialized as BCS.
+  optional Bcs bcs = 1;
+
+  // The digest of this TransactionEffects.
+  optional string digest = 2;
+
+  // Version of this TransactionEffects.
+  optional Version version = 3;
+
+  // The status of the execution.
+  optional ExecutionStatus status = 4;
+
+  // The epoch when this transaction was executed.
+  optional uint64 epoch = 5;
+
+  // The gas used by this transaction.
+  optional GasCostSummary gas_used = 6;
+
+  // The transaction digest.
+  optional string transaction_digest = 7;
+
+  // Information about the gas object. Also present in the `changed_objects` vector.
+  //
+  // System transaction that don't require gas will leave this as `None`.
+  optional ChangedObject gas_object = 8;
+
+  // The digest of the events emitted during execution,
+  // can be `None` if the transaction does not emit any event.
+  optional string events_digest = 9;
```

Comment:
Do we return the events themselves anywhere except for a transaction that was just executed or as part of the `FullCheckpointData`?

Reply:
So events are included in the ExecutedTransaction message which contains all Transaction information (as well as balance changes)

Reply:
There's a world where maybe we embed those fields here vs there though

Reply:
Got it -- I hadn't spotted that `GetTransactionResponse` contains an `ExecutedTransaction` -- all good.